ilastik_scripts/ilastik_startup.py

```diff
@@ -138,9 +143,110 @@ def find_library(name):
     ctypes.util.find_library = find_library
 
 
+def check_parent_for_dir_recursive(path: pathlib.Path, dir_name: str) -> Optional[pathlib.Path]:
+    if path.is_dir() and any(p.name == dir_name for p in path.iterdir()):
+        return path
```

Comment:
This would also return a `Path` if it came across a file (not a dir) named `dir_name`.